server/src/test/java/org/elasticsearch/common/io/stream/BytesStreamsTests.java

```diff
@@ -586,6 +587,23 @@ public void testWriteMapOfLists() throws IOException {
         out.close();
     }
 
+    public void testWriteMapAsList() throws IOException {
+        final int size = randomIntBetween(0, 100);
+        final Map<String, String> expected = Maps.newMapWithExpectedSize(size);
+        for (int i = 0; i < size; ++i) {
+            final String value = randomAlphaOfLength(5);
+            expected.put(value, value);
+        }
+
+        final BytesStreamOutput out = new BytesStreamOutput();
+        out.writeMapAsList(expected, StreamOutput::writeString);
+        final StreamInput in = StreamInput.wrap(BytesReference.toBytes(out.bytes()));
+        final Map<String, String> loaded = in.readMapFromList(StreamInput::readString, Function.identity());
```

Comment:
Could we have a nontrivial `keyMapper` in the test, e.g. take the first couple of characters of the string?

Reply:
>take the first couple of characters of the string

This could become flaky in case test will generate multiple strings with a common prefix. 
I think it is possible to add some constant to the key instead..